### PR TITLE
ROX-12429 Support debugging deployed Fleet components via VSCode

### PR DESCRIPTION
**Pull request by mtesseract**
## Description

This PR adds functionality for debugging fleet-manager and fleetshard-sync in VSCode while they are running as deployed Pods in the Kubernetes test environment.

This requires the [Cloud Code extension](https://marketplace.visualstudio.com/items?itemName=GoogleCloudTools.cloudcode).

There have been changes in the Makefile, in the Dockerfiles and there is a new file `.vscode/launch.json` which contains the debugging configuration.

Let's begin with the Dockerfiles.

For supporting debuggable pods we need to extend the build recipe (e.g. to compile with certain flags and include the `dlv` debugger). In order to not maintain a dedicated `Dockerfile.debug` I decided to implement this requirement with the multi-target feature of Dockerfiles. One can now use `docker build --target [standard|debug] ...` to build a non-debuggable image or a debuggable image. Furthermore, the new Dockerfile has the complete build process containerized (as it should actually be in most situations). In order to preserve the previous behaviour for building images I have renamed the old `Dockerfile` to `Dockerfile.hybrid`, because it contains somewhat of an hybrid building approach: binaries built natively on the host are just copied into a new image. Due to Go not providing a reliable way for caching built dependencies, the Makefile target `image/build` still uses the hybrid approach under the hood, which is much faster.

The target `image/build/local`, which is used by the dev env scripts uses new new Dockerfile and the Makefile does not respect the environment variables `DEBUG_IMAGE` (and also `IGNORE_REPOSITORY_DIRTINESS`). If `DEBUG_IMAGE=true`, this means that  the new make target `full-image-tag` produces a tag of the form `fleet-manager-dbg:...` instead of `fleet-manager:...` and `image/build/local` will create a debuggable image.

In the dev scripts I had to make the commands to be executed in the containers configurable.

## Checklist (Definition of Done)
- [x] ~~Unit and integration tests added~~ (n/a)
- [x] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [x] CI and all relevant tests are passing
- [x] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [x] ~~Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.~~ (n/a)

## Test manual

1. `make image/build` works as before.
2. `DEBUG_IMAGE=true make image/build/local` builds a debug image (name suffixed with `-dbg`)
3. `DEBUG_IMAGE=true  make full-image-tag` outputs a debug image tag.
4. `IGNORE_REPOSITORY_DIRTINESS=true make full-image-tag` does not add a `-dirty` suffix.
5. `DEBUG_PODS=true dev/env/scripts/up.sh` builds and deploys debuggable images.
6. Debugging via VSCode works:
    (a) Install Cloud Code extension.
    (b) Deploy using `DEBUG_PODS=true dev/env/scripts/up.sh`
    (c) Start the debugging using the "Attach to Fleet-Manager Pod" configuration.
    (d) Set breakpoint, e.g. at https://github.com/stackrox/acs-fleet-manager/blob/mc%2Fk8s-debugging/internal/dinosaur/pkg/handlers/dinosaur.go#L138
    (e) Query the list of centrals, e.g.: `OCM_TOKEN=$STATIC_TOKEN ./scripts/fmcurl "rhacs/v1/centrals"`
    (f) Verify that the debugger stopped at breakpoint and that variables can be inspected.

### CODE REVIEW

**Review by johannes94 — CHANGES_REQUESTED** [review]
LGTM after the requested changed are done

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
```
[the remark this answers is not in the document]
Comment (mtesseract) [reply .vscode/launch.json:4]: added

File: Makefile [hunk 2]
```diff
@@ -498,14 +508,34 @@ docker/login/internal:
...
-# Build the binary and image
+# Build the image in a hybrid fashion, i.e. building binaries directly on the host leveraging
+# Go's cross-compilation capabilities and then copying these binaries into a new Docker image.
 image/build: GOOS=linux
 image/build: GOARCH=amd64
-image/build: IMAGE_REF="$(external_image_registry)/$(image_repository):$(image_tag)"
+image/build: IMAGE_REF ?= "$(external_image_registry)/$(image_repository):$(image_tag)"
 image/build: fleet-manager fleetshard-sync
-	DOCKER_CONFIG=${DOCKER_CONFIG} $(DOCKER) build -t $(IMAGE_REF) .
+	DOCKER_CONFIG=${DOCKER_CONFIG} $(DOCKER) build -t $(IMAGE_REF) -f . Dockerfile.hybrid
```
[the remark this answers is not in the document]
Comment (mtesseract) [reply Makefile]: fixed

**Review by johannes94 — APPROVED** [review]
LGTM

**Pull request merged**